src/main/java/nl/uu/cs/ape/domain/BioToolsAPI.java

```diff
@@ -172,86 +195,122 @@ private static JSONArray fetchToolsFromURI(String url) throws JSONException, IOE
-						apeInputFormats.put(bioType.getString("uri"));
+				try {
+					apeInputs = calculateBioToolsInputOutput(function.getJSONArray("input"),
+							bioJsonTool.getString("biotoolsID"));
+					apeJsonTool.put("inputs", apeInputs);
+				} catch (BioToolsAnnotationException e) {
+					if (excludeBadAnnotation) {
+						notAcceptedOperations++;
+						continue;
 					}
-					apeInput.put("format_1915", apeInputFormats);
-
-					apeInputs.put(apeInput);
 				}
-				apeJsonTool.put(ToolAnnotationTag.INPUTS.toString(), apeInputs);
-
-				// reading outputs
 				JSONArray apeOutputs = new JSONArray();
-				JSONArray bioOutputs = function.getJSONArray("output");
-				// for each output
-				for (int j = 0; j < bioOutputs.length(); j++) {
-
-					JSONObject bioOutput = bioOutputs.getJSONObject(j);
-					JSONObject apeOutput = new JSONObject();
-					JSONArray apeOutputTypes = new JSONArray();
-					JSONArray apeOutputFormats = new JSONArray();
-					// add all data types
-					for (JSONObject bioType : APEUtils.getListFromJson(bioOutput, "data", JSONObject.class)) {
-						apeOutputTypes.put(bioType.getString("uri"));
-					}
-					apeOutput.put("data_0006", apeOutputTypes);
-					// add all data formats
-					for (JSONObject bioType : APEUtils.getListFromJson(bioOutput, "format", JSONObject.class)) {
-						apeOutputFormats.put(bioType.getString("uri"));
+				try {
+					apeOutputs = calculateBioToolsInputOutput(function.getJSONArray("output"),
+							bioJsonTool.getString("biotoolsID"));
+					apeJsonTool.put("outputs", apeOutputs);
+				} catch (BioToolsAnnotationException e) {
+					if (excludeBadAnnotation) {
+						notAcceptedOperations++;
+						continue;
 					}
-					apeOutput.put("format_1915", apeOutputFormats);
-
-					apeOutputs.put(apeOutput);
 				}
-				apeJsonTool.put(ToolAnnotationTag.OUTPUTS.toString(), apeOutputs);
-
-				apeToolsAnnotations.put(apeJsonTool);
+				if (!excludeBadAnnotation ||
+				(apeInputs.length() > 0 && apeOutputs.length() > 0)) {
+					apeJsonTool.put("biotoolsID", bioJsonTool.getString("biotoolsID"));
+					apeToolsAnnotations.put(apeJsonTool);
+				}
 			}
 		}
+		log.info("Provided bio.tools: " + bioToolsAnnotation.length());
+		log.info("Total bio.tools functions: " + bioToolFunctions);
+		log.info("Errored bio.tools functions: " + notAcceptedOperations);
+		log.info("Created APE annotations: " + apeToolsAnnotations.length());
 
 		return new JSONObject().put("functions", apeToolsAnnotations);
 	}
 
+
+
+
+	
+	/**
+	 * Method converts input and output tool annotations, following bio.tools schema,
+	 * into the APE library tool annotation format.
+	 * @param bioInputs - JSONArray with the bio.tools input/output annotations.
+	 * @param toolID - ID of the tool that is being converted.
+	 * 
+	 * @return JSONArray with the APE library input/output annotations.
+	 */
+	private static JSONArray calculateBioToolsInputOutput(JSONArray bioInputs, String toolID) {
```

Comment:
Missing throws declaration. Is that by purpose?

Reply:
Good catch, I forgot to update the documenatation after adding it!